pkg/clusterregistryconfig/flags.go

```diff
@@ -0,0 +1,349 @@
+	blockedRegistries          []string
+	insecureRegistries         []string
+	allowedRegistriesForImport string
+	platformAllowlist          string
+	additionalTrustedCa        string
+}
+
+func AddClusterRegistryConfigFlags(cmd *cobra.Command) *ClusterRegistryConfigArgs {
+	args := &ClusterRegistryConfigArgs{}
+
+	cmd.Flags().StringSliceVar(
+		&args.allowedRegistries,
+		allowedRegistriesFlag,
+		nil,
+		"A comma-separated list of registries for which image pull and push actions are allowed.",
+	)
+
+	cmd.Flags().StringSliceVar(
+		&args.insecureRegistries,
+		insecureRegistriesFlag,
+		nil,
+		"A comma-separated list of registries which do not have a valid TLS certificate or only support HTTP connections.",
+	)
+
+	cmd.Flags().StringSliceVar(
+		&args.blockedRegistries,
+		blockedRegistriesFlag,
+		nil,
+		"A comma-separated list of registries for which image pull and push actions are denied.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.allowedRegistriesForImport,
+		allowedRegistriesForImportFlag,
+		"",
+		"Limits the container image registries from which normal users can import images. "+
+			"The format should be a comma-separated list of 'domainName:insecure'. "+
+			"'domainName' specifies a domain name for the registry. "+
+			"'insecure' indicates whether the registry is secure or insecure.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.platformAllowlist,
+		platformAllowlistFlag,
+		"",
+		"A reference to the id of the list of registries that needs to be whitelisted for the platform to work. "+
+			"It can be omitted at creation and updating and its lifecycle can be managed separately if needed.",
+	)
+	cmd.Flags().MarkHidden("platform-allowlist")
+
+	cmd.Flags().StringVar(
+		&args.additionalTrustedCa,
+		additionalTrustedCaFlag,
+		"",
+		"A json file containing the registry hostname as the key,"+
+			" and the PEM-encoded certificate as the value, for each additional registry CA to trust.")
+
+	return args
+}
+
+func GetClusterRegistryConfigArgs(args *ClusterRegistryConfigArgs) (
+	[]string, []string, []string, string, string) {
+	return args.allowedRegistries, args.blockedRegistries,
+		args.insecureRegistries, args.additionalTrustedCa, args.allowedRegistriesForImport
+}
+
+func GetClusterRegistryConfigOptions(cmd *pflag.FlagSet,
+	args *ClusterRegistryConfigArgs, isHostedCP bool) (
+	*ClusterRegistryConfigArgs, error) {
+
+	if args != nil && !isHostedCP {
+		return nil, fmt.Errorf("Setting the registry config is only supported for hosted clusters")
+	}
+
+	result := &ClusterRegistryConfigArgs{}
+
+	result.allowedRegistries = args.allowedRegistries
+	result.insecureRegistries = args.insecureRegistries
+	result.blockedRegistries = args.blockedRegistries
+	result.additionalTrustedCa = args.additionalTrustedCa
+	result.allowedRegistriesForImport = args.allowedRegistriesForImport
+	result.platformAllowlist = args.platformAllowlist
+
+	if !IsClusterRegistryConfigSetViaCLI(cmd) {
+		if !interactive.Enabled() {
+			return nil, nil
+		}
+	}
```

Comment:
This could be simplified
```suggestion
	if !IsClusterRegistryConfigSetViaCLI(cmd) &&  !interactive.Enabled() {
			return nil, nil
	}
```

Reply:
Updated :)